src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,345 +5,132 @@
+        auto tensor = get_tensor(it);
+        auto type = tensor.get_element_type();
+        if (req->get_tensor(it).data(type) != tensor.data(type)) {
+            req->set_tensor(it, ov::Tensor(it, tensor.data()));
+        }
     }
 }
 
-void SyncInferRequest::CopyInputsIfNeeded() {
-    for (const auto& it : _networkInputs) {
-        auto& name = it.first;
+void SyncInferRequest::copy_inputs_if_needed() {
+    for (const auto& it : get_inputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        CopyBlobIfNeeded(GetBlob(name), m_batched_request_wrapper._inferRequestBatched->GetBlob(name), true);
+        auto dst_tensor = m_batched_request_wrapper->_infer_request_batched->get_tensor(it);
+        copy_tensor_if_needed(get_tensor(it), dst_tensor, true);
     }
 }
 
-void SyncInferRequest::CopyBlobIfNeeded(InferenceEngine::Blob::CPtr src, InferenceEngine::Blob::Ptr dst, bool bInput) {
-    auto bufferDst = dst->buffer();
-    auto ptrDst = bufferDst.as<char*>();
-    auto bufferSrc = src->cbuffer();
-    auto ptrSrc = bufferSrc.as<const char*>();
-    ptrdiff_t szDst = dst->byteSize();
-    ptrdiff_t szSrc = src->byteSize();
+void SyncInferRequest::copy_tensor_if_needed(const ov::Tensor& src, ov::Tensor& dst, bool bInput) {
+    auto ptrDst = static_cast<char*>(dst.data());
+    auto ptrSrc = static_cast<char*>(src.data());
+    ptrdiff_t szDst = dst.get_byte_size();
+    ptrdiff_t szSrc = src.get_byte_size();
     if (bInput) {
         ptrdiff_t offset = szSrc != szDst ? m_batch_id * szDst / m_batch_size : 0;
-        if ((ptrDst + offset) == ptrSrc)
+        if ((ptrDst + offset) == ptrSrc) {
             return;
-        else
+        } else
             memcpy(ptrDst + offset, ptrSrc, szSrc);
     } else {
         ptrdiff_t offset = szSrc != szDst ? m_batch_id * szSrc / m_batch_size : 0;
-        if ((ptrSrc + offset) == ptrDst)
+        if ((ptrSrc + offset) == ptrDst) {
             return;
-        else
+        } else
             memcpy(ptrDst, ptrSrc + offset, szDst);
     }
 }
 
-void SyncInferRequest::CopyOutputsIfNeeded() {
-    for (const auto& it : _networkOutputs) {
-        auto& name = it.first;
+void SyncInferRequest::copy_outputs_if_needed() {
+    for (const auto& it : get_outputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        CopyBlobIfNeeded(m_batched_request_wrapper._inferRequestBatched->GetBlob(name), GetBlob(name), false);
+        auto dst_tensor = get_tensor(it);
+        copy_tensor_if_needed(m_batched_request_wrapper->_infer_request_batched->get_tensor(it), dst_tensor, false);
     }
 }
+
+void SyncInferRequest::infer() {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::vector<std::shared_ptr<ov::IVariableState>> SyncInferRequest::query_state() const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::vector<ov::ProfilingInfo> SyncInferRequest::get_profiling_info() const {
+    OPENVINO_NOT_IMPLEMENTED;
```

Comment:
why these methods are not implemented?
should we just re-call methods of underlying HW plugin?